yarn/src/main/scala/org/apache/spark/scheduler/cluster/YarnClientSchedulerBackend.scala

```diff
@@ -61,6 +60,9 @@ private[spark] class YarnClientSchedulerBackend(
     // to the executors
     super.start()
 
+    if (conf.contains("spark.yarn.credentials.file")) {
+      YarnSparkHadoopUtil.get.startExecutorDelegationTokenRenewer(conf)
```

Comment:
This looks a little out of place to the uninitiated, since this code is running in the driver. IIUC, what's happening in client mode is that the AM is still doing the relogin / token renewal, and now the driver is just monitoring HDFS, like other executors, to get the updated tokens.

Is that correct? Could you add a comment explaining that if so?